openshift/insights-operator
# Expose Insights recommendations as alerts

This implements the feature of exposing the Insights recommendations as Prometheus alerts. Note that this feature is **disabled by default** (as described in the [OCP enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/insights/prometheus-alerts-for-insights-recommendations.md#goals). You need to create **`enableInsightsAlerts`** attribute in the `support` secret in the `openshift-config` namespace to enable it!

**Manual testing steps**
Launch a cluster-bot cluster on this PR.
Copy and run the oc login command for said cluster.
Run the following commands.
```
oc scale --namespace=openshift-cluster-version deployment/cluster-version-operator --replicas 0
oc new-project project1
oc new-project project2
oc patch netnamespace project1 --type=merge -p '{"egressIPs":["192.168.1.99"]}'
oc patch netnamespace project2 --type=merge -p '{"egressIPs":["192.168.1.99"]}'
oc patch storage.operator.openshift.io cluster --type merge --patch '{"spec":{"managementState": "Unmanaged"}}'
oc create secret generic support --namespace=openshift-config --from-literal=endpoint=https://cloud.redhat.com/api/ingress/v1/upload --from-literal=interval=10m
oc scale -n openshift-insights deployment/insights-operator --replicas 0
oc scale -n openshift-insights deployment/insights-operator --replicas 1
```
Check the Observe > Alerting tab in OpenShift Web Console. You should see 3 instances of InsightsRecommendationActive alert there (they will be pending for the first 5 minutes).

## Categories



## Sample Archive


No new data

## Documentation

This will require an official OCP documentation

## Unit Tests


This will require integration testing

## Privacy


Yes. There are no sensitive data in the newly collected information.

## Changelog


## Breaking Changes

No

## References


https://issues.redhat.com/browse/CCXDEV-7327
https://bugzilla.redhat.com/show_bug.cgi?id=???
https://access.redhat.com/solutions/???
